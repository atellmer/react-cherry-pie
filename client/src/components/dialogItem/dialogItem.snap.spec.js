import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import muiTheme from '../../config/theme';
import TmDialogItem from './index';

const props = {
  interlocutor: {},
  changeInterlocutor: () => {},
  dialog: {
    user: {
      id: '12345',
      name: {
        first: 'Alex',
        last: 'Plex'
      },
      avatar: {
        thumbnail: 'path/to/avatar'
      },
      online: false
    },
    message: {
      value: {
        text: 'lorem ipsum'
      },
      status: {
        delivered: true,
        read: false,
        new: false
      },
      timestamp: new Date()
    }
  }
};

jest.mock('react-dom');

test('Jest: TmDialogItem (Snapshot)', () => {
  const component = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <TmDialogItem {...props}/>
    </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
