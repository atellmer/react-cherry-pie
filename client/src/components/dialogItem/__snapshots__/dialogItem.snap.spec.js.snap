exports[`test Jest: TmDialogItem (Snapshot) 1`] = `
<div
  className=""
  onClick={[Function]}>
  <div
    className={undefined}>
    <div
      className="">
      <img
        className={undefined}
        size={48}
        src="path/to/avatar"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": "50%",
            "color": "#ffffff",
            "display": "inline-flex",
            "fontSize": 24,
            "height": 48,
            "justifyContent": "center",
            "muiPrepared": true,
            "userSelect": "none",
            "width": 48,
          }
        } />
    </div>
    <div
      className="">
      <div
        className={undefined}>
        <div
          className="">
          Alex Plex
        </div>
        <div
          className="">
          lorem ipsum
        </div>
        <div
          className={undefined}>
          <svg
            className={undefined}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24">
            <path
              d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
          </svg>
          <div
            className={undefined}>
            Fri Jan 20 2017
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
